packages/babel-plugin-ember-test-metadata/utils.js

```diff
@@ -33,26 +34,76 @@ function getNodeProperty(node, path) {
 }
 
 /**
- * Get a normalized file path. If Embroider prefix is present, strip it out
+ * Function to parse defaults.project and only return info to be used by the plugin
+ * @param {object} project Ember defaults.project
+ * @returns {object} Contains project name and ember-addon path info
+ */
+function getProjectInfo(project) {
+  const parsedProjectInfo = {};
+
+  if (project) {
+    parsedProjectInfo.pkg = {
+      name: project.pkg.name,
+      'ember-addon': {
+        paths: project.pkg['ember-addon'].paths,
+      },
+    };
+  }
+
+  return parsedProjectInfo;
+}
+
+/**
+ * Strip out project name prefix segement(s) from file path
+ * @param {array} tokens File path segments
+ * @param {object} projectInfo Contains project name, ember-addon path info
+ * @returns {string} A joined file path, e.g. tests/acceptance/my-test.js
+ */
+function getParsedClassicFilepath(tokens, projectInfo) {
+  const projectNameTokens = projectInfo.pkg.name.split(path.sep);
+
+  tokens.splice(0, tokens.indexOf(projectNameTokens[0]) + projectNameTokens.length);
+
+  return tokens.join(path.sep);
+}
+
+/**
+ * Strip out Embroider prefix (embroider/nnnnnn) segements from file path
+ * @param {array} tokens File path segments
+ * @returns {string} A joined file path, e.g. tests/acceptance/my-test.js
+ */
+function getParsedEmbroiderFilepath(tokens) {
+  const RELATIVE_PATH_ROOT = 2;
+
+  tokens.splice(0, tokens.lastIndexOf(EMBROIDER) + RELATIVE_PATH_ROOT);
+
+  return tokens.join(path.sep);
+}
+
+/**
+ * Get a normalized file path, based on whether the app build is classic or with Embroider
  * @param {object} fileOpts Babel state.file.opts which include root and filename props
+ * @param {object} projectInfo Contains project name, ember-addon path info
  * @returns {string} E.g. tests/acceptance/my-test.js
  */
-function getNormalizedFilePath(fileOpts) {
+function getNormalizedFilePath(fileOpts, projectInfo) {
   let { root, filename } = fileOpts;
   const tokens = filename.split(path.sep);
-  const EMBROIDER = 'embroider';
+  const isEmbroider = tokens.includes(EMBROIDER);
 
-  if (tokens.includes(EMBROIDER)) {
-    const RELATIVE_PATH_ROOT = 2;
-
-    tokens.splice(0, tokens.lastIndexOf(EMBROIDER) + RELATIVE_PATH_ROOT);
+  if (isEmbroider) {
+    filename = getParsedEmbroiderFilepath(tokens);
```

Comment:
not sure if it matters, but this might be easier to reason about this if it were inlined.

```javascript
const { filename } = fileOpts;
// const embroiderBuild = ???
// const classicBuild = ???

let normalizedPath = filename;

if (embroiderBuild) {
  const pathParts = filename.split(path.sep);
  const extendedParts = 2;
  const separatorStart = pathParts.lastIndexOf('embroider');
  pathParts.splice(0, separatorStart + extendedParts)
  normalizedPath = path.join(pathParts);
} else if (classicBuild) {
  const { name } = projectInfo.pkg;
  normalizedPath = normalizedPath.slice(name.length);
}

return path.relative(root, filename);
```

Reply:
I gave the exact opposite advice. Combining both is harder to reason about IMO. I do think it's fine to have a single entry point, and to split up the logic between those in separate functions. That's what I'd indicated in my prior comment.

Reply:
I appreciate both perspectives. In this case, splitting them out to smaller functions made it more readable to me, easier to grok, and unit-testable. I think eventually we will have a 3rd function that these functions will use, that will be for handling in-repo addon cases.